File: app/src/main/java/com/gas/app/https/GankItemService.java
package com.gas.app.https;


import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import com.gas.app.bean.GankItemData;
import com.gas.app.bean.HttpResult;

/**
 * Author: Othershe
 * Time: 2016/8/12 16:50
 */

public interface GankItemService {

    @GET("{suburl}")
    Observable<HttpResult<List<GankItemData>>> getGankItemData(@Path("suburl") String suburl);


}

File: app/src/main/java/com/gas/app/https/GankItemService.kt
package com.gas.app.https

import com.gas.app.bean.GankItemData
import com.gas.app.bean.HttpResult
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Author: Othershe
 * Time: 2016/8/12 16:50
 */
interface GankItemService {
    @GET("{suburl}")
    fun getGankItemData(@Path("suburl") suburl: String?): Observable<HttpResult<List<GankItemData>>>
}